addons/website_sale_loyalty/models/website.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class Website(models.Model):
+    _inherit = 'website'
+
+    def _default_loyalty_program(self):
+        return self.env['loyalty.program'].search([], limit=1)
+
+    has_loyalty = fields.Boolean("Has Loyalty Program", help="Enables a loyalty program for this website", default=False)
+    loyalty_id = fields.Many2one('loyalty.program', string='Loyalty Program', help='The loyalty program used by this website', default=_default_loyalty_program)
+
+    @api.onchange('has_loyalty')
+    def _onchange_has_loyalty(self):
+        if self.has_loyalty:
+            self.loyalty_id = self._default_loyalty_program()
+        else:
+            self.loyalty_id = False
+
+    def contains_discount_reward(self, product_id):
+        """Returns true if the loyalty program contains the specified product_id as a discount reward"""
+        if self.has_loyalty:
+            return self.loyalty_id.contains_discount_reward(product_id)
+        return False
```

Comment:
This is so imperative-style code...
```suggestion
        return self.has_loyalty and self.loyalty_id.contains_discount_reward(product_id)
```